website/project/views/node.py

```diff
@@ -1003,6 +1003,66 @@ def get_children(auth, node, **kwargs):
     return _render_nodes(nodes, auth)
 
 
+def node_child_tree(user, node_ids):
+    """ Format data to test for node privacy settings for use in treebeard.
+    """
+    items = []
+    addons = []
+    for node_id in node_ids:
+        node = Node.load(node_id)
+        assert node, '{} is not a valid Node.'.format(node_id)
+
+        can_read = node.has_permission(user, 'read')
```

Comment:
We show the entire tree to the user, and don't allow modifications if they don't have access, so read access is appropriate in this case.